.github/workflows/unit-tests.yml

```diff
@@ -12,6 +12,7 @@ jobs:
       - name: Install Python 3
         uses: actions/setup-python@v3
         with:
+          cache: 'pip' # caching pip dependencies
```

Comment:
While you're changing these, can you update to [v4.2.0](https://github.com/actions/setup-python/releases/tag/v4.2.0) and use `check-latest: true`?